### PR TITLE
feat: PI-1233 subdomain filtering using environment variables

### PR DESCRIPTION
**Pull request by pirey**

### CODE REVIEW

File: pkg/auth/iam/iam.go [hunk 1]
```diff
@@ -60,6 +62,11 @@ type ErrorResponse struct {
 
 // NewFilter creates new Filter instance
 func NewFilter(client iam.Client) *Filter {
+	enableSubdomainFiltering, _ := strconv.ParseBool(os.Getenv(constant.EnvEnableSubdomainFiltering))
```
Comment (rexydwybrundy) [inline pkg/auth/iam/iam.go:65]: Can we make it consistent using the LookUpEnv so we also able to handle if env not exists also?

Something like this
`if s, exists := os.LookupEnv("FULL_ACCESS_LOG_ENABLED"); exists {
		value, err := strconv.ParseBool(s)
		if err != nil {
			logrus.Errorf("Parse FULL_ACCESS_LOG_ENABLED env error: %v", err)
		}
		FullAccessLogEnabled = value
	}`

**Review by rexydwybrundy — APPROVED** [review]

**Comment by medreza** [comment]
what do you think if we rename the env to `SUBDOMAIN_VALIDATION_ENABLED `?

**Comment by pirey** [comment]
updated.

File: pkg/auth/iam/iam.go [hunk 2]
```diff
@@ -60,6 +62,16 @@ type ErrorResponse struct {
 
 // NewFilter creates new Filter instance
 func NewFilter(client iam.Client) *Filter {
+	if s, exists := os.LookupEnv(constant.EnvSubdomainValidationEnabled); exists {
```
Comment (marselsampe) [inline pkg/auth/iam/iam.go:65]: Suggest to make it something like this below, 
So we could reuse it when we have another Filter option in the future:

```
        options := &FilterInitializationOptions{}        	
	if s, exists := os.LookupEnv(constant.EnvSubdomainValidationEnabled); exists {
		enableSubdomainFiltering, err := strconv.ParseBool(s)
		if err != nil {
			logrus.Errorf("Parse %s env error: %v", constant.EnvSubdomainValidationEnabled, err)
		}
		if enableSubdomainFiltering {
			options.AllowSubdomainMatchRefererHeaderValidation: true
		}
	}
	
	return &Filter{iamClient: client, options: options}
```

Comment (pirey) [reply pkg/auth/iam/iam.go:66]: updated

File: pkg/auth/iam/iam.go [hunk 3]
```diff
@@ -60,7 +62,17 @@ type ErrorResponse struct {
...
 func NewFilter(client iam.Client) *Filter {
-	return &Filter{iamClient: client, options: &FilterInitializationOptions{}}
+	options := &FilterInitializationOptions{}
+	if s, exists := os.LookupEnv(constant.EnvSubdomainValidationEnabled); exists {
+		subdomainValidationEnabled, err := strconv.ParseBool(s)
+		if err != nil {
+			logrus.Errorf("Parse %s env error: %v", constant.EnvSubdomainValidationEnabled, err)
+		}
+		if subdomainValidationEnabled == true {
+			options = &FilterInitializationOptions{AllowSubdomainMatchRefererHeaderValidation: true}
```
Comment (marselsampe) [inline pkg/auth/iam/iam.go:72]: `options.AllowSubdomainMatchRefererHeaderValidation = true`

Comment (pirey) [reply pkg/auth/iam/iam.go]: updated.

**Review by marselsampe — APPROVED** [review]

**Review by angelo-chan — APPROVED** [review]

**Pull request merged**